fastapi_sqla/base.py

```diff
@@ -20,6 +20,31 @@
 _ENGINE_KEYS_REGEX = re.compile(r"fastapi_sqla__(?!_)(.+)(?<!_)__(?!_).+")
 
 
+async def startup():
+    engine_keys = _get_engine_keys()
+    engines = {key: sqla.new_engine(key) for key in engine_keys}
+    for key, engine in engines.items():
+        if not _is_async_dialect(engine):
+            sqla.startup(key=key)
+        else:
+            await async_sqla.startup(key=key)
+
+
+def setup_middlewares(app: FastAPI):
+    engine_keys = _get_engine_keys()
+    engines = {key: sqla.new_engine(key) for key in engine_keys}
+    for key, engine in engines.items():
+        if not _is_async_dialect(engine):
+            app.middleware("http")(
+                functools.partial(sqla.add_session_to_request, key=key)
+            )
+        else:
+            app.middleware("http")(
+                functools.partial(async_sqla.add_session_to_request, key=key)
+            )
+
+
+# TODO: Remove this function once all FastAPI apps stops using startup/shutdown events
```

Comment:
Since this is a OOS lib, should we use the [deprecated](https://github.com/tantale/deprecated) lib and add a deprecated decorator here to tell consumers to upgrade to the new startup, because it'll be deprecated in v4?

Reply:
done